sui_programmability/adapter/src/adapter.rs

```diff
@@ -902,6 +902,12 @@ fn struct_tag_equals_sig_token(
         SignatureToken::StructInstantiation(idx, args) => {
             struct_tag_equals_struct_inst(module, function_type_arguments, arg_type, *idx, args)
         }
+        SignatureToken::TypeParameter(_) => type_tag_equals_sig_token(
+            module,
+            function_type_arguments,
+            &TypeTag::Struct(arg_type.clone()),
+            param_type,
+        ),
```

Comment:
@lxfind this match branch is missing and it's causing type check to fail for move function with TypeParameter in the method signature e.g. `fn<T>(T)`, can you check if this is the right fix?

Reply:
@tnowacki could you help take a look at this? I am not quite familiar with this part of the code.

Reply:
These functions are setup to avoid clones, which may or may not be overkill. But keeping with that style, this should be 
```
SignatureToken::TypeParameter(idx) => match &function_type_arguments[*idx as usize] {
    TypeTag::Struct(s) => arg_type == s,
    _ => false,
},
```